packages/cursorless-engine/src/api/CursorlessEngineApi.ts

```diff
@@ -39,3 +40,13 @@ export interface CommandApi {
    */
   runCommandSafe(...args: unknown[]): Promise<unknown>;
 }
+
+export interface CommandRunnerDecorator {
+  /**
+   * @param commandRunner: A CommandRunner.
+   * @param readableHatMap: A ReadOnlyHatMap.
+   * @returns A new CommandRunner that invokes the provided CommandRunner in
+   *   addition to performing some other work.
+   */
+  wrapCommandRunner: (readableHatMap: ReadOnlyHatMap, commandRunner: CommandRunner) => CommandRunner;
```

Comment:
It feels wrong that all CommandRunnerDecorators should have to take a ReadOnlyHatMap. However, it is generated inside runCommand asynchronously from hatTokenMap and I did not want to have to generate it again inside the decorator.  Suggestions welcome.